indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,236 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import timeTableURL from 'indico-url:event_registration.api_event_timetable';
+
+import PropTypes from 'prop-types';
+import React, {useState, useEffect} from 'react';
+import {useSelector} from 'react-redux';
+import {Checkbox, Form} from 'semantic-ui-react';
+
+import {
+  FinalCheckbox,
+  FinalRadio,
+  FinalField,
+  FinalInput,
+  validators as v,
+} from 'indico/react/forms';
+import {useIndicoAxios} from 'indico/react/hooks';
+import {Translate} from 'indico/react/i18n';
+
+import '../../../styles/regform.module.scss';
+import {getStaticData} from '../selectors';
+
+function TimetableSessionsComponent({
+  value,
+  name,
+  sessionData,
+  onChange,
+  allowFullDays,
+  display,
+  ...props
+}) {
+  function handleValueChange(e, data) {
+    const id = data.value;
+    const newValue = value.includes(id) ? value.filter(el => el !== id) : [...value, id];
+    onChange(newValue);
+  }
+
+  return (
+    <Form.Group styleName="timetablesessions-field">
+      {sessionData &&
+        Object.keys(sessionData)?.map((date, index) => (
+          <SessionBlockHeader
+            value={value}
+            index={index}
+            name={name}
+            data={sessionData[date]}
+            label={date}
+            allowFullDays={allowFullDays}
+            display={display}
+            key={`session-block-${index}`}
+            handleValueChange={handleValueChange}
+            onChange={onChange}
+            props={props}
+          />
+        ))}
+    </Form.Group>
+  );
+}
+
+TimetableSessionsComponent.propTypes = {
+  value: PropTypes.arrayOf(PropTypes.number).isRequired,
+  name: PropTypes.string.isRequired,
+  sessionData: PropTypes.object,
+  onChange: PropTypes.func.isRequired,
+  allowFullDays: PropTypes.bool.isRequired,
+  display: PropTypes.string.isRequired,
+};
+
+TimetableSessionsComponent.defaultProps = {
+  sessionData: {},
+};
+
+function SessionBlockHeader({
+  value,
+  data,
+  label,
+  allowFullDays,
+  display,
+  handleValueChange,
+  onChange,
+}) {
+  const [isChecked, setIsChecked] = useState(false);
+  const [isIndeterminate, setIsIndeterminate] = useState(false);
+  const [isExpanded, setIsExpanded] = useState(display === 'expanded');
+
+  const arrowLabel = isExpanded ? '▼' : '▶';
+  const isExpandedClass = isExpanded ? '' : 'hidden';
+  const childrenIds = data.map(e => e.id);
+
+  function handleHeaderClick() {
+    setIsExpanded(!isExpanded);
+  }
+
+  function handleHeaderChange() {
+    const newValue =
+      isChecked || isIndeterminate
+        ? value.filter(element => !childrenIds.includes(element))
+        : [...value, ...childrenIds];
+    onChange(newValue);
+  }
+
+  useEffect(() => {
+    setIsChecked(() =>
```

Comment:
Should I add a comment here explaining is checked if all children are checked and is indeterminated if some children are checked?